### ISSUE TITLE
Seeking advice on localization

### ISSUE THREAD

**Issue by wzleong** [BVLC/caffe#4631]
I am starting to work on localization problem using Caffe. 
I managed to train my network for classification. For localization, I am planning to use regression. 
1. In the deploy.prototxt, is there a layer needed for regression? I understand that during training, the loss layer type is defined as Euclidean loss. 
Previously when I train for classification, I have to define a layer type "softmax" for the deploy.prototxt, whereas in the train.prototxt we have a loss layer of "softmaxwithloss". 
I guess for regression, we don't require a separate layer to wok on regression (to compute the result, unlike for softmax we have to apply some maths because the scores are treated as exponential un-normalized probability) because the Fully Connected layer has trained weights to perform the regression and output the results desired?
For localization, my ultimate goal is to obtain the 4 bounding boxes, so does it mean that my final FC layer is 1x1x4, where each of the output is the bounding box data. 

2. For classification, I input my data's using the datum message. But I understand that the datum message does not have additional options to store the bounding box data. Is there an easy work around for this? Or I have to code my own input data message in the proto file?  

Thanks 
Wei Zhen

**Comment by shelhamer** [comment]
Please ask modeling questions and advice on our mailing list.

From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:

> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.
